### DISCUSSION TITLE
A way to check if a `str` is actually a hashed password

### DISCUSSION THREAD

**Discussion by Diegovsky** [frankie567/pwdlib#6] [Ideas]
I noticed that with `argon2` at least you can know if a `str` actually represents a hashed password. It has a specific format that encodes this information.

I'm currently writing a custom type decorator for SQLAlchemy that hashes a password if it is unhashed and to prevent already hashed passwords from being rehashed, it would be nice to have a way (say, a `pwdlib.is_hashed` function) to check if it at least looks like a hashed password from pwdlib.

If course, I can just tag my passwords with something like a `hashed:` prefix, but I think this is seriously redundant considering pwd already has this information encoded.

What do you think?

**Comment by frankie567** [comment]
Interesting! Actually, we already have `identify` methods on each implemented algorithms, and we use it to determine the right one to apply when verifying a password:

https://github.com/frankie567/pwdlib/blob/17fd0c6a2f4b18cdc5fe6d8115844b7aea4cbfce/pwdlib/_hash.py#L85-L87

The main wrapper doesn't expose it, but I suppose we could add it. Something like:

```py
class PasswordHash:
    # ...
    def identify(
        self, hash: typing.Union[str, bytes]
    ) -> HasherProtocol:
        for hasher in self.hashers:
            if hasher.identify(hash):
                return hasher
        raise exceptions.UnknownHashError(hash)
```

If you want to take a stab at a PR, you're very welcome 🙏

**Reply by Diegovsky** [reply]
I would like to avoid making a PR because I'm not familiar with the codebase (and contributing to python repositories in general), and it looks to me you already have a solution in mind, which fits my needs. Can't this be committed as-is?
